lib/package-managers/npm-proxy.js

```diff
@@ -39,10 +39,11 @@ const noProxyCheck = (scope) => {
 };
 
 if (proxyUrl) {
-    const proxy = global.URL ? new URL(proxyUrl) : url.parse(proxyUrl);
```

Comment:
Thanks for the PR. The legacy url API is [deprecated](https://nodejs.org/docs/latest/api/url.html#url_legacy_url_api), so we'd need a code-path handling both `auth` from legacy url, and `username`:`password` from WHATWG URL to support future versions.